src/mag_toolkit/calibration/Layer.py

```diff
@@ -22,11 +25,32 @@ class Layer(BaseModel, ABC):
     rotation: list[list[list[float]]] | None = None
 
     @classmethod
-    def from_file(cls, path: Path):
+    def from_file(cls: type[T], path: Path) -> T:
         with open(path) as fid:
             as_dict = yaml.safe_load(fid)
         model = cls(**as_dict)
-        return model
+
+        # If data is defined in a separate file load it, otherwise
+        # leave it as is.
+        if not model.metadata.data_filename:
+            return model
+
+        files_to_try: list[Path] = [
+            model.metadata.data_filename,
+            path.parent / model.metadata.data_filename.name,
+        ]
+
+        for file in files_to_try:
+            if file.exists():
+                return cls._load_data_file(file, model)
```

Comment:
Why is this on the class? Couln't it just be an instance method called once you have instantiated a layer object? Would be simpler and no need to pass around model object?